src/renderer/kube-object.store.ts

```diff
@@ -106,23 +131,20 @@ export abstract class KubeObjectStore<T extends KubeObject = any> extends ItemSt
   }
 
   @action
-  async loadAll(namespaces: string[] = []): Promise<void> {
+  async loadAll(namespaces?: string[], { replace = false /*partial update*/ } = {}): Promise<void> {
```

Comment:
I'm not completely sold on the `replace` parameter idea here. Could we just always remove all items that match given namespaces before we call `loadItems()`? By doing that we would always have a fresh data for given namespaces and store would still keep other namespaces in memory.

```ts
      if (this.api.isNamespaced) {
        // clear any existing items from given namespaces
        const items = this.items.toJS();

        const leftItems = items.filter(item => !namespaces.includes(item.getNs()));

        this.items.replace(leftItems);
      } else {
        this.items.replace([]);
      }

      const items = await this.loadItems({ namespaces, api: this.api });
```

Reply:
We cannot replace _all items_ in store when loading partial namespaces. (since store is singleton and store from the extension will override items). By clearing items from just loaded namespaces is a good idea.

Reply:
Reverted logic back as was in 4.0. 
Now it replaces all items in the store and reflects with `NamespaceSelect` state.